### ISSUE TITLE
Help me setting up model definition file

### ISSUE THREAD

**Issue by duckroll** [uber/ludwig#598]
I have a dataset thats consists of 2 users with matching interests:
user1
"cars, psychology, politics" 
user2
"beauty, philosophy, sport"

I have tried sequence2sequence from examples but accuracy doesn't go above 0.02. I have also tried to tweak numerous output and input features. Sorry for asking it here but stackoverflow seems dead for ludwig questions.

**Comment by msaisumanth** [comment]
What's your current model definition? Try ```set``` feature for the output.

**Comment by duckroll** [comment]
Here is my model definition: 

input_features:
    -
        name: user1
        type: sequence
        level: word
        encoder: rnn
        cell_type: lstm
        reduce_output: null

output_features:
    -
        name: user2
        type: set
        level: word
        decoder: tagger
        cell_type: lstm
        attention: bahdanau

training:
    batch_size: 128

And these are results (2 last epochs):

╒═════════╤═══════════╤═══════════╕
│ user2   │      loss │   jaccard │
╞═════════╪═══════════╪═══════════╡
│ train   │ 1093.5133 │    0.0669 │
├─────────┼───────────┼───────────┤
│ vali    │ 1466.8972 │    0.0530 │
├─────────┼───────────┼───────────┤
│ test    │ 1452.4886 │    0.0499 │
╘═════════╧═══════════╧═══════════╛
Last improvement of loss on combined happened 4 epochs ago

╒═════════╤═══════════╤═══════════╕
│ user2   │      loss │   jaccard │
╞═════════╪═══════════╪═══════════╡
│ train   │ 1451.6767 │    0.0709 │
├─────────┼───────────┼───────────┤
│ vali    │ 1615.1785 │    0.0541 │
├─────────┼───────────┼───────────┤
│ test    │ 1695.9522 │    0.0503 │
╘═════════╧═══════════╧═══════════╛
Last improvement of loss on combined happened 5 epochs ago

**Comment by msaisumanth** [comment]
It's hard to help without really knowing the data and the task. But, first of all the metric here is [Jaccard index](https://en.wikipedia.org/wiki/Jaccard_index), not accuracy. 

I don't know the number of different classes in your outputs, so not sure what's a good Jaccard index to aim for.

**Comment by duckroll** [comment]
I tweaked my settings a little bit and this is the beast I got:

╒═════════╤═══════════╤═══════════╕
│ user2   │      loss │   jaccard │
╞═════════╪═══════════╪═══════════╡
│ train   │   68.8706 │    0.8622 │
├─────────┼───────────┼───────────┤
│ vali    │ 1757.2357 │    0.0480 │
├─────────┼───────────┼───────────┤
│ test    │ 1586.6935 │    0.0477 │
╘═════════╧═══════════╧═══════════╛

Training jaccard index is pretty high but the validation is still low. Do you think increasing my dataset size can improve it?

**Comment by msaisumanth** [comment]
Yes, but your model is overfitting.

**Comment by duckroll** [comment]
I can deal with overfitting by feeding my model more data it seems.

**Comment by duckroll** [comment]
Is there any ludwig option to avoid overfitting?

**Comment by msaisumanth** [comment]
Overfitting is a very common and generic problem in Machine Learning. More data would certainly help.

I'm closing this issue as there's no Ludwig specific problem. Feel free to create another issue if you hit a roadblock :)

**Comment by duckroll** [comment]
Thanks for help
